Parser/Python.asdl

```diff
@@ -39,7 +39,7 @@ module Python
           | Match(expr subject, match_case* cases)
 
           | Raise(expr? exc, expr? cause)
-          | Try(stmt* body, excepthandler* handlers, stmt* orelse, stmt* finalbody)
+          | Try(stmt* body, excepthandler* handlers, stmt* orelse, stmt* finalbody, int star)
```

Comment:
IMHO it makes sense in terms of static analyzers / linters to have this in a separate node (just like the distinction between `FunctionDef`/`AsyncFunctionDef`) since the tools that consume `ast.Try` do not expect this behavioural change. Perhaps `Try` / `TryStar`. What do you think @pablogsal?

Reply:
Yeah, I think this should be Indeed a separate node. It will help a lot tools consuming AST nodes and it will help separating concerns in the compiler.

Reply:
Sure, I'll split them in the AST. 

Whether to separate them in the compiler is a good question, I considered doing that (based on the star field on the Try node) but I haven't done it so far because there is a significant amount of shared code between except and except*. I am now considering to refactor the except implementation a bit so that all cleanup is outside the loop (rather than cleanup and then go to end). That would make the two flows overlap even more. Or we can decide to separate them completely as you suggest.

Reply:
I just pushed a commit that splits the AST nodes to Try/TryStar. 

This pretty much forced me to split the compiler code as well, because we need to know whether we are working on s->v.Try or s->v.TryStar (which are unrelated structs).   They have the same structure so I could cast one to the other, but this feels wrong.  I don't mind too much the split of compiler_try_except, but we do have now compiler_try_finally and compiler_try_star_finally which contain exactly the same logic.  And a few pieces of identical validation code as well.

Let me know what you think.